### DISCUSSION TITLE
Allow setting heading tag as HTML tag for containers and widgets

### DISCUSSION THREAD

**Discussion by AF1QUE** [elementor/elementor#25005] [Feature Request]
### Prerequisites

- [X] I have searched for similar feature requests in both open and closed discussions and cannot find an existing request.
- [X] I have verified that the feature is still missing from the latest stable versions of Elementor or Elementor Pro.

### What problem is your feature request going to solve? Please describe..

Problem: There is no option to pick heading tags like (h1,h2,h3,h4,h5,h6) while changing the HTML tag option from a container. Please take a look at the image below for better understanding.

![Screenshot 2024-02-03 at 5 44 46 AM](https://github.com/elementor/elementor/assets/46385375/0fd9fbb8-4399-43d4-9091-2dcfa5b86559)

### Describe the solution you'd like.

Solution: It would be nice if there's a option for picking up header tags from the HTML tag selection field.

### Agreement

- [X] I agree that my request may be closed without action if it doesn't meet all the requirements.

**Comment by portalzine** [comment]
For anyone else searching for a solution. Its not valid HTML. h2 is an inline container and only allows phrasing content :)

```
add_action( 'elementor/element/container/section_layout/after_section_end', function( $element, $args ) {
    $element->update_control( 'html_tag', [
        'options' => [
            'div'     => 'div',
            'header'  => 'header',
            'footer'  => 'footer',
            'main'    => 'main',
            'article' => 'article',
            'section' => 'section',
            'aside'   => 'aside',
            'nav'     => 'nav',
            'h2'      => 'h2',
            'h3'      => 'h3',
            'h4'      => 'h4',
        ],
    ] );
}, 10, 2 );
```

or merge

```
add_action( 'elementor/element/container/section_layout/after_section_end', function( $element, $args ) {
    $control = $element->get_controls( 'html_tag' );
    if ( ! $control ) return;

    $existing_options = $control['options'] ?? [];

    $element->update_control( 'html_tag', [
        'options' => array_merge( $existing_options, [
            'h2' => 'h2',
            'h3' => 'h3',
            'h4' => 'h4',
        ] ),
    ] );
}, 10, 2 );
```
